### DISCUSSION TITLE
White Screen When Launching App Without Display of Images in Asset

### DISCUSSION THREAD

**Discussion by ndgsghdj** [sst-inc/help#53] [Q&A - iOS] [answered]
Launching the app, the images in the Assets folder would not be displayed on the app
<img width="364" alt="Screenshot 2021-10-28 at 3 39 49 PM" src="https://user-images.githubusercontent.com/75611248/139209336-5384ae07-3f39-48ac-96c4-6cc3f6fb4d18.png">
Furthermore, when pressing the "Add" ("+") button to add pictures, it seems that I get an error like this:
<img width="865" alt="Screenshot 2021-10-28 at 3 41 07 PM" src="https://user-images.githubusercontent.com/75611248/139209506-0179590f-fa9b-4e31-ab89-49892fd12ec5.png">
I had expected the image to display as I had put an element in the Photos array in Photo.swift with an image name filled out as well as the image being put in the Assets folder. However, it still seems to not display.

**Comment by Joeispro99** [comment, marked as answer]
Can you send the xcode file? @ndgsghdj

**Reply by Joeispro99** [reply]
Im pretty sure its a sigabrt error so i need to check the connections inspector of the image virew

**Reply by ndgsghdj** [reply]
[photo gallery.zip](https://github.com/sst-inc/help/files/7438735/photo.gallery.zip)
here is the xcode file

**Reply by Joeispro99** [reply]
There shouldn't be 2 classes of the same file. You declared a class on both line 12 and 141. Delete line 141 to 144 and add the delegates for the navigation controller and image picker controller. In addition, you should put the code properly. Functions should not be put in view did load and instead, outside of that view did load function. Not only that, in the numberOfItemsInSection function,you should not return zero and instead, the number of images in the image array. LAstly, add constraints on your collection view cell or else it would end up like
<img width="358" alt="Screenshot 2021-10-29 at 5 48 14 PM" src="https://user-images.githubusercontent.com/65899074/139414358-543cc43c-ae97-45b8-92f3-5f43dfceaa30.png">

refer to slide 19 in iOS Lesson 5 on the constraints https://docs.google.com/presentation/d/1k16mNd9lUCUlMzqqarB9CurlZBIo4egLyDhgVN-8TFM/edit#slide=id.gfab7606283_0_11

if you need help with constraints, reply and ping this thread.

[photo gallery.zip](https://github.com/sst-inc/help/files/7440885/photo.gallery.zip)

**Reply by ndgsghdj** [reply]
> There shouldn't be 2 classes of the same file. You declared a class on both line 12 and 141. Delete line 141 to 144 and add the delegates for the navigation controller and image picker controller. In addition, you should put the code properly. Functions should not be put in view did load and instead, outside of that view did load function. Not only that, in the numberOfItemsInSection function,you should not return zero and instead, the number of images in the image array. LAstly, add constraints on your collection view cell or else it would end up like <img alt="Screenshot 2021-10-29 at 5 48 14 PM" width="358" src="https://user-images.githubusercontent.com/65899074/139414358-543cc43c-ae97-45b8-92f3-5f43dfceaa30.png">
> 
> refer to slide 19 in iOS Lesson 5 on the constraints https://docs.google.com/presentation/d/1k16mNd9lUCUlMzqqarB9CurlZBIo4egLyDhgVN-8TFM/edit#slide=id.gfab7606283_0_11
> 
> if you need help with constraints, reply and ping this thread.
> 
> [photo gallery.zip](https://github.com/sst-inc/help/files/7440885/photo.gallery.zip)

how do I get the constraints menu in slide 19? @Joeispro99

**Reply by granwyntan** [reply]
Hi @ndgsghdj, 
If you're referring to the constraints section of the Size Inspector, this might help
<img width="1440" alt="Screenshot 2021-10-29 at 21 22 56" src="https://user-images.githubusercontent.com/61010627/139443317-397d313b-a7c7-472a-807a-422754b1feb3.png">
To create constraints, I would recommend using this which is on the bottom left of ur main storyboard
<img width="283" alt="Screenshot 2021-10-29 at 21 23 10" src="https://user-images.githubusercontent.com/61010627/139443304-80d01bfa-2f55-41bd-ba40-23add2ca7fc5.png">
Hope this helps

**Reply by ndgsghdj** [reply]
ah I see. Thanks.
